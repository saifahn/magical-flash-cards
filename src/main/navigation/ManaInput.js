import React, { Component } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { media, text } from '../../utils/theme';


const Input = styled('input')`
  font-size: ${text.md};
  font-family: 'Lora', serif;
  text-align: center;
  border: none;
  background: rgba(255, 255, 255, 0.8);
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
  padding: 0.5rem;
  margin: 0 auto 0.75rem;
  width: 100%;

  ${media.sm`
    font-size: ${text.lg};
  `}

  ${media.md`
    font-size: ${text.xl};
  `}

  ${media.lg`
    max-width: 560px;
  `}

  &:focus {
    outline: none;
  }
`;

export default class ManaInput extends Component {
  onManaChange = (e) => {
    const { setManaFilter } = this.props;
    setManaFilter(e.target.value);
  }

  render() {
    const placeholder = 'type a mana cost e.g. WWBBB';
    return (
      <Input
        type="text"
        onChange={this.onManaChange}
        placeholder={placeholder}
      />
    );
  }
}

ManaInput.propTypes = {
  setManaFilter: PropTypes.func.isRequired,
};
