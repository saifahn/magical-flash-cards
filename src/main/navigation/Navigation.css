.c-navigation {
  /* background-color: #D3AD97; */
  /* flex-basis: 20%; */
}