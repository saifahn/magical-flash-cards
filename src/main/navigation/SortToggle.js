import React, { Component } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { media, text } from '../../utils/theme';

export const ToggleButton = styled('button')`
  background: none;
  border: none;
  font-size: ${text.md};
  padding: 0 0.5rem;
  font-family: 'Lora', serif;
  position: relative;
  outline: none;
  color: ${props => (props.isSelected ? 'black' : 'rgba(0, 0, 0, 0.6)')};

  :before {
    content: "";
    position: absolute;
    width: 50%;
    bottom: 1rem;
    border-bottom: ${props => (props.isSelected ? '2px solid black' : 0)};

    ${media.sm`
        bottom: 1.25rem;
        border-bottom: ${props => (props.isSelected ? '2.5px solid black' : 0)};
    `}
  }

  ${media.sm`
    font-size: ${text.lg};
    margin: 0;
    padding: 0 0.5rem;
  `}
`;

export default class SortToggle extends Component {
  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    const { toggleSort, sorter } = this.props;
    toggleSort(sorter);
  }

  render() {
    const { sorter, children, sortBy } = this.props;
    const isSelected = (sortBy.indexOf(sorter) > -1);
    return (
      <ToggleButton
        onClick={this.handleClick}
        className="c-sort-button"
        isSelected={isSelected}
      >
        {children}
      </ToggleButton>
    );
  }
}

SortToggle.propTypes = {
  toggleSort: PropTypes.func.isRequired,
  sorter: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  sortBy: PropTypes.arrayOf(PropTypes.string).isRequired,
};
