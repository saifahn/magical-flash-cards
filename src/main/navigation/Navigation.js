import React from 'react';
import './Navigation.css';
import ManaInput from './ManaInput';
import Colors from './Colors';

const Sorter = props => (
  <button
    onClick={props.sortByCMC}
    className="c-button"
  >
    Sort by CMC
  </button>
);

const ColorSorter = props => (
  <button
    onClick={props.sortByColor}
    className="c-button"
  >
    Sort by Color
  </button>
);

const Navigation = props => (
  <section className="c-navigation">
    <h3>This is where the navigation goes</h3>
    <Colors onClick={props.onClick} />
    <ManaInput setManaFilter={props.setManaFilter} />
    <Sorter sortByCMC={props.sortByCMC} />
    <ColorSorter sortByColor={props.sortByColor} />
    <button
      onClick={props.toggleGrid}
      className="c-button"
    >
      Grid Toggle
    </button>
  </section>
);

export default Navigation;
