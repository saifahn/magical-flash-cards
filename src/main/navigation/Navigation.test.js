/* eslint-env jest */
import React from 'react';
import { shallow, mount } from 'enzyme';
import Navigation from './Navigation';
import ManaInput from './ManaInput';
import SortToggle from './SortToggle';

let wrapper;
let toggleSortSpy;
let sets;

beforeEach(() => {
  sets = []
  toggleSortSpy = jest.fn();
});

test('it renders correctly', () => {
  wrapper = shallow(<Navigation sets={sets} />);
  expect(wrapper).toMatchSnapshot();
});

test('it renders a ManaInput component', () => {
  wrapper = shallow(<Navigation sets={sets} />);
  expect(wrapper.find(ManaInput).length).toBe(1);
});

test('it renders two SortToggle components', () => {
  wrapper = shallow(<Navigation sets={sets} />);
  expect(wrapper.find(SortToggle).length).toBe(2);
});

test('clicking colour sorter calls toggleSort with colour', () => {
  wrapper = mount(<Navigation sets={sets} toggleSort={toggleSortSpy} />);
  wrapper.find({ sorter: 'colour' }).simulate('click');
  expect(toggleSortSpy).toHaveBeenLastCalledWith('colour');
});

test('clicking cmc sorter calls toggleSort with cmc', () => {
  wrapper = mount(<Navigation sets={sets} toggleSort={toggleSortSpy} />);
  wrapper.find({ sorter: 'cmc' }).simulate('click');
  expect(toggleSortSpy).toHaveBeenLastCalledWith('cmc');
});
