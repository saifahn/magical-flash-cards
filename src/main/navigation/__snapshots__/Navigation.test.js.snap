// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<section
  className="c-navigation"
>
  <ManaInput />
  <SortToggle
    sorter="cmc"
  />
  <SortToggle
    sorter="colour"
  />
  <button
    className="c-button"
  >
    Grid Toggle
  </button>
  <SetSelector />
</section>
`;
